### PR TITLE
add configuration to hide dust payments

### PR DESCRIPTION
**Pull request by piyalbasu**
Screenshot 2024-10-15 at 11 57 53 AM
Screenshot 2024-10-15 at 11 58 03 AM
Screenshot 2024-10-15 at 11 58 16 AM

### CODE REVIEW

File: @shared/api/internal.ts [hunk 1]
```diff
@@ -1176,11 +1176,11 @@ export const saveAllowList = async ({
 export const saveSettings = async ({
   isDataSharingAllowed,
   isMemoValidationEnabled,
-  isNonSSLEnabled,
+  isHideDustEnabled,
```
Comment (piyalbasu) [inline @shared/api/internal.ts:1179]: there was a bug where, i think through a bad merge, the `isNonSSLEnabled` configuration was in both the Preferences and in the Advanced Settings. It used to be in Preferences but was later moved to Advanced Settings.

Taking this opportunity to replace errant `isNonSSLEnabled` config with the new `isHideDustEnabled`

File: extension/src/popup/views/AccountHistory/index.tsx [hunk 2]
```diff
@@ -99,13 +104,25 @@ export const AccountHistory = () => {
           operation.type ===
             Horizon.HorizonApi.OperationResponseType.createAccount &&
           operation.account !== publicKey;
+        const isDustPayment =
+          isPayment &&
+          "asset_type" in operation &&
+          operation.asset_type === "native" &&
+          "to" in operation &&
+          operation.to === publicKey &&
```
Comment (piyalbasu) [inline extension/src/popup/views/AccountHistory/index.tsx:112]: checking for a native payment to the user's account

Comment (aristidesstaffieri) [reply extension/src/popup/views/AccountHistory/index.tsx:112]: should this be flipped? `operation.asset_type === "native"`

I think dust payments would typically not be xlm right, as the spam is to try to get bunk tokens to show up in the interface by sending dust amounts of them.

File: extension/src/popup/views/Preferences/index.tsx [hunk 3]
```diff
@@ -15,34 +15,35 @@ import "./styles.scss";
...
   };
 
   const handleSubmit = async (formValue: SettingValues) => {
     const {
       isValidatingMemoValue,
       isDataSharingAllowedValue,
-      isNonSSLEnabledValue,
+      isHideDustEnabledValue,
     } = formValue;
+    console.log(formValue);
```
Comment (aristidesstaffieri) [inline extension/src/popup/views/Preferences/index.tsx:39]: stray log

File: extension/src/popup/views/__tests__/AccountHistory.test.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        }}
+      >
+        <AccountHistory />
+      </Wrapper>,
+    );
+
+    await waitFor(() => screen.getByTestId("AccountHistory"));
+    expect(screen.getByTestId("AccountHistory")).toBeDefined();
+    const historyNodes = screen.getAllByTestId("history-item");
+    expect(historyNodes.length).toEqual(2);
```
Comment (aristidesstaffieri) [inline extension/src/popup/views/__tests__/AccountHistory.test.tsx:72]: along with the inferred length of the history item without the dust item, should we explicitly check that the correct history item is filtered out?

Comment (piyalbasu) [reply extension/src/popup/views/__tests__/AccountHistory.test.tsx:72]: Good call! I'll add

File: extension/src/popup/views/AccountHistory/index.tsx (hunk 2 above)
Comment (piyalbasu) [reply extension/src/popup/views/AccountHistory/index.tsx:112]: Actually, according to this [doc](https://docs.google.com/document/d/1tpGsVx3ZlRZbdhedkOfj_cUE2Ixdzct1v9xZP9zj3FE/edit), we're specifically on the lookout for small XLM payments

Comment (aristidesstaffieri) [reply extension/src/popup/views/AccountHistory/index.tsx:112]: oh cause of the memo, yup got it!

**Review by aristidesstaffieri — APPROVED** [review]

**Pull request merged**